agenta-backend/agenta_backend/services/evaluators_service.py

```diff
@@ -607,9 +607,18 @@ async def auto_contains_json(
     settings_values: Dict[str, Any],  # pylint: disable=unused-argument
     lm_providers_keys: Dict[str, Any],  # pylint: disable=unused-argument
 ) -> Result:
-    if not isinstance(output, str):
-        output = output.get("data", "")
     try:
+        if not isinstance(output, str):
+            # Attempt to retrieve 'data' key from output if it's a dictionary
+            output = output.get("data", "") if isinstance(output, dict) else output
```

Comment:
Why do we check twice for 'data' ?
At this stage, this would be `output.data.data`, which shouldn't be the case.

Checking twice for `not isinstance(output, str)` is ok though.
The first time if for `output: str`, the second is for `output.data: str`.

Reply:
This makes sense, thank you.

Reply:
The second time isn't enough. The behaviour is to ensure that the value of data is a string of JSON, and not a dictionary. If it is a dictionary, raise an exception.

The below example will result in an exception because the retrieved "data" value is still a dictionary.

```python
output = {"data": {"generation": '{"message": "The capital of Nigeria is Abuja."}'}, "trace": []}
```

Whereas, the second example below proceeds without issues because the retrieved "data" value is a string.

```python
output = {"data": '{"message": "The capital of Nigeria is Abuja."}', "trace": []}
```